torch/_dynamo/variables/builtin.py

```diff
@@ -807,6 +807,10 @@ def _dyn_proxy(self, tx, *args, **kwargs):
     def _call_iter_tuple_list(self, tx, obj=None, *args, **kwargs):
         if self._dynamic_args(*args, **kwargs):
             return self._dyn_proxy(tx, *args, **kwargs)
+        if isinstance(obj, variables.IteratorVariable):
+            # For non-list iterators, we will guard on vars that
+            # determine the control flow
```

Comment:
hmm, I wonder if this is true?

As long as you guard on all python values and control flow is not influenced by tensor values.